### PR TITLE
Send Room updates when participant counts change

### PR DESCRIPTION
**Pull request by davidzhao**
Reduces the number of unneeded generation with ProtoProxy

### CODE REVIEW

**Review by paulwe — APPROVED** [review]
lgtm!

File: pkg/rtc/room.go [hunk 1]
```diff
@@ -47,9 +48,10 @@ type broadcastOptions struct {
 type Room struct {
 	lock sync.RWMutex
 
-	protoRoom *livekit.Room
-	internal  *livekit.RoomInternal
-	Logger    logger.Logger
+	protoRoom  *livekit.Room
```
Comment (boks1971) [inline pkg/rtc/room.go:51]: Is this still needed? Can `protoProxy` be the one true value?

File: pkg/rtc/room.go [hunk 2]
```diff
@@ -760,11 +753,9 @@ func (r *Room) createJoinResponseLocked(participant types.LocalParticipant, iceS
 	}
 
 	return &livekit.JoinResponse{
-		Room:              r.protoRoom,
+		Room:              r.ToProto(),
 		Participant:       participant.ToProto(),
 		OtherParticipants: otherParticipants,
-		ServerVersion:     r.serverInfo.Version,
-		ServerRegion:      r.serverInfo.Region,
```
Comment (boks1971) [inline pkg/rtc/room.go:767]: Are these not used anymore?

File: pkg/rtc/room.go (hunk 1 above)
Comment (davidzhao) [reply pkg/rtc/room.go:51]: yes they are. protoProxy is only holding a cached copy. it's not designed to be modified directly.

File: pkg/rtc/room.go (hunk 2 above)
Comment (davidzhao) [reply pkg/rtc/room.go:767]: they have been deprecated for awhile. we've been dual writing them into both ServerInfo and also at root

**Pull request merged**